### PR TITLE
Limit evaluators

### PR DESCRIPTION
**Pull request by merose**
Add limit evaluation to the realtime plugin. Refactor to reduce duplicate code.

This is in fulfillment of VIPERGDS-127.

### CODE REVIEW

File: src/providers/realtime-telemetry-provider.js [hunk 1]
```diff
@@ -23,12 +23,107 @@
...
+            }
+            return obj
+        }
+    }
+
+    /*
+     * Adds limit range information to an object based on the monitoring
+     * result.
+     */
+    addLimitRange(datum, result, obj) {
```
Comment (akhenry) [inline src/providers/realtime-telemetry-provider.js:96]: Maybe we could rename `obj` to `evaluationResult` or something? I was a little confused initially about what this was, and thought at first it was an Open MCT domain object.

File: src/providers/realtime-telemetry-provider.js [hunk 2]
```diff
@@ -29,6 +29,105 @@ import {
...
+                cssClass: MONITORING_RESULT_CSS[datum.monitoringResult],
+                name: datum.monitoringResult
+            }
+            if (datum.rangeCondition
+                && datum.rangeCondition in RANGE_CONDITION_CSS) {
+                obj.cssClass += ' '
+                obj.cssClass += RANGE_CONDITION_CSS[datum.rangeCondition]
+                this.addLimitRange(datum, datum.monitoringResult, obj)
+            }
+            console.log('result -> ' + JSON.stringify(obj))
```
Comment (akhenry) [inline src/providers/realtime-telemetry-provider.js]: Let's remove this debugging code.

File: src/providers/realtime-telemetry-provider.js [hunk 3]
```diff
@@ -88,6 +195,7 @@ export default class RealtimeTelemetryProvider {
 
         this.socket.onmessage = (event) => {
             let data = JSON.parse(event.data);
+            console.log('socket.onmessage event=' + JSON.stringify(data))
```
Comment (akhenry) [inline src/providers/realtime-telemetry-provider.js]: Let's remove this debugging output

**Review by akhenry — CHANGES_REQUESTED** [review]
@merose This looks really great. I've just included some small minor suggestions.

File: src/providers/realtime-telemetry-provider.js (hunk 1 above)
Comment (merose) [reply src/providers/realtime-telemetry-provider.js:96]: Will fix.

File: src/providers/realtime-telemetry-provider.js (hunk 2 above)
Comment (merose) [reply src/providers/realtime-telemetry-provider.js]: Fixed.

File: src/providers/realtime-telemetry-provider.js (hunk 3 above)
Comment (merose) [reply src/providers/realtime-telemetry-provider.js]: Fixed.

**Pull request merged**